src/helpers/spacewalk.ts

```diff
@@ -87,3 +87,12 @@ export function calculateDeadline(
   const end = now.plus({ seconds: remainingDurationSecs });
   return end;
 }
+
+export function estimateRequestCreationTime(
+  activeBlockOpenTime: number,
+  blockTimeSec = 12,
+) {
+  const secondsAgo = activeBlockOpenTime * blockTimeSec;
+  const now = DateTime.now();
+  return now.minus({ seconds: secondsAgo })
```

Comment:
This would not really work as expected. The problem is that `activeBlockOpenTime` is not related to the current 'epoch' at the time but it's just some number that is incremented starting from 0 when the blockchain started. Thus, we can't relate it to `DateTime` that easily. 

I think the following should work
```
export function estimateRequestCreationTime(
  activeBlockOpenTime: number,
  currentActiveBlock: number,
  blockTimeSec = 12,
) {
  const activeBlocksPassed = currentActiveBlock - activeBlockOpenTime;
  const secondsAgo = activeBlocksPassed * blockTimeSec;
  const now = DateTime.now();
  return now.minus({ seconds: secondsAgo })
}
```

Reply:
Uhm I see, but how often is the `activeBlockOpenTime` incremented? If it's a counter that gets incremented per block? If that is the case, then it's exactly the same as saying that the it started N blocks ago. From your code I would understand that `activeBlockOpenTime` is exactly the block number when the request was created. Then, subtracting the current block number minus the block number when the request started, you indeed get an amount of blocks. So `activeBlockOpenTime` is a `BlockNumber`. Is that correct ?

Reply:
Okay, that is correct according to [this](https://github.com/interlay/interbtc/blob/b174d63da831cadff70c117633c8a0cb4af790cc/primitives/src/lib.rs#L129). I am still getting some weird dates using your code. I am getting the current active block from getActiveBlockNumber() in useSecurityPallet()

Reply:
> So activeBlockOpenTime is a BlockNumber. Is that correct? 

yes exactly

What are the values you pass for `activeBlockOpenTime` and `currentActiveBlock`, did you log them and can maybe share them to get a better understanding?